admin/services/check-window-v2.service.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const moment = require('moment')
+const monitor = require('../helpers/monitor')
+const checkWindowDataService = require('./data-access/check-window.data.service')
+
+const checkWindowV2Service = {}
+
+/**
+ * Get all check windows recorded in the database with their status
+ * @returns {Array} List of check windows
+ */
+checkWindowV2Service.getCheckWindows = async () => {
+  const checkWindows = await checkWindowDataService.sqlFindCheckWindowsWithStatus()
+  checkWindows.forEach(cw => (cw.canRemove = cw.status === 'Inactive'))
+  return checkWindows
+}
+
+/**
+ * Mark check window as deleted based on url slug
+ * @returns {Object|Error} Either a successful message or throws an exception
+ */
+checkWindowV2Service.markDeleted = async (urlSlug) => {
+  const currentUTCTimestamp = moment.utc()
```

Comment:
feels like there should be validation here to ensure that `urlSlug` is actually a valid UUID.

Reply:
@GuyHarwood introduced validation but I am guessing we could extend this validation in other areas where we need to validate `urlSlug` fields.

Reply:
i did think that earlier.  it will no doubt need adding to the pupil edit area, and maybe some others.